src/Magento/FunctionalTestingFramework/Module/MagentoWebDriver.php

```diff
@@ -826,6 +834,32 @@ public function makeScreenshot($name = null)
         AllureHelper::addAttachmentToCurrentStep($screenName, 'Screenshot');
     }
 
+    /**
+     * Check connectivity to running selenium server
+     *
+     * @return void
+     * @throws TestFrameworkException
+     */
+    public function checkSeleniumServerReadiness()
+    {
+        try {
+            $driver = RemoteWebDriver::create(
+                $this->wdHost,
+                $this->capabilities,
+                $this->connectionTimeoutInMs,
+                $this->requestTimeoutInMs,
+                $this->httpProxy,
+                $this->httpProxyPort
+            );
+            $driver->close();
+        } catch (WebDriverCurlException $e) {
+            throw new TestFrameworkException(
+                "Can't connect to Webdriver at {$this->wdHost}.\n"
+                . "Please make sure that Selenium Server is running."
+            );
+        }
```

Comment:
Why don't you include all of this in `TroubleshootClass` instead of creating separate method designed for very small concrete task?

Reply:
This is to shortcut the initialization routines. Alternatively, we have to have a test to initiate the routines which I personally don't prefer.